client/src/api/plugin/queries.tsx

```diff
@@ -0,0 +1,150 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import axios from 'axios';
+import { PluginDefinition, PluginEndpoint, PluginParameters } from '../Models';
+
+export function useGetPlugins() {
+  return useQuery<PluginDefinition[]>(
+    ['plugins'],
+    async () => {
+      const response = await axios.get<PluginDefinition[]>('/api/plugins/').catch((err) => {
+        console.error('Failing getting plugins from api', err);
+        if (import.meta.env.IQENGINE_PLUGINS) {
+          const plugins = JSON.parse(import.meta.env.IQENGINE_PLUGINS);
+          return { data: plugins };
+        } else {
+          return { data: [] };
+        }
+      });
+      return response.data;
+    },
+    {
+      placeholderData: [],
+    }
+  );
+}
+
+export function useGetPlugin(plugin: PluginDefinition) {
+  return useQuery<string[]>(
+    ['plugin', plugin.name],
+    async () => {
+      const response = await axios.get<string[]>(plugin.url);
+      // make sure we onlyt return in case of a string array
+      if (!Array.isArray(response.data)) throw new Error(`Plugin ${plugin.name} does not return a string array`);
+      return response.data;
+    },
+    {
+      enabled: !!plugin,
+    }
+  );
+}
+
+export function useGetPluginParameters(pluginUrl: string) {
+  return useQuery<PluginParameters>(
+    ['plugin', 'parameters', pluginUrl],
+    async () => {
+      const response = await axios.get<PluginParameters>(pluginUrl);
+      return response.data;
+    },
+    {
+      enabled: !!pluginUrl,
+    }
+  );
+}
+
+export function useGetPluginDetailed(plugin: PluginDefinition) {
+  return useQuery<PluginEndpoint>(
+    ['plugin', plugin.name, 'detailed'],
+    async () => {
+      // make sure we have a trailing slash  to concatenate the url
+      if (plugin.url && !plugin.url.endsWith('/')) plugin.url += '/';
+      const response = await axios.get<string[]>(plugin.url);
+      if (!response.data) throw new Error(`Plugin ${plugin.name} does not return correctly`);
+      const result = new PluginEndpoint();
+      result.name = plugin.name;
+      result.url = plugin.url;
+      result.plugins = {};
+      for (const item of response.data) {
+        const pluginUrl = new URL(item, plugin.url);
+        const pluginParameters = await axios.get<PluginParameters>(pluginUrl.toString()).catch((err) => {
```

Comment:
should we really be doing this at this point? surely it would be better to only do this when we open the modal?

Reply:
We could have another query that would do this when we do open the modal actually we could have this query as part of the modal since prior to the modal no information in this query it is needed.

Reply:
just thinking that it's better to not fetch the data before is needed

Reply:
Agreed will change the query to just be enabled is the modal is opened that way would avoid to fetch before it is needed

Reply:
Changed